HTML Cheat Sheet

jsstrn opened the pull request:
**What does your Instant Answer do?**

It lists the HTML elements.

**What is the data source for your Instant Answer? (Provide a link if possible)**

[HTML element reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element) from Mozilla Developer Network.

**Why did you choose this data source?**

I chose this because they listed HTML elements in a logical structure, alphabetized, and separated deprecated elements.

**Are there any other alternative (better) data sources?**

Others that I've looked at contained deprecated elements or only listed common HTML elements. There's a lot of outdated sources out there.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Web developers. People learning HTML.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Yes. See [here](https://duck.co/ideas/idea/6160/html-cheat-sheet).

**Are you having any problems? Do you need our help with anything?**

All is well. I'm definitely open to feedback and suggestions for improvement.

**Where did you hear about DuckDuckHack? (For first time contributors)**

TL;DR: Serendipity.

I've been a long time user of DuckDuckGo and found out about the DuckDuckHack open source projects after reading about its privacy policy and the "Internet bubble effect".

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

![screen shot 2015-08-30 at 2 45 41 am](https://cloud.githubusercontent.com/assets/1199611/9563757/53fc43e8-4ec1-11e5-927c-7b81a7dd98f2.png)

![screen shot 2015-08-30 at 2 46 00 am](https://cloud.githubusercontent.com/assets/1199611/9563760/5d8dbe50-4ec1-11e5-9fe3-6ef8cdd4f3fa.png)

IA Page: https://duck.co/ia/view/html_cheat_sheet

MariagraziaAlastra commented:
Hey @jsstrn, thanks for your contribution!
We'll review this and get back to you soon :)

moollaza commented on share/goodie/cheat_sheets/json/html.json:
```diff
@@ -0,0 +1,341 @@
+{
+  "id": "html_cheat_sheet",
+  "name": "HTML Elements",
+  "description": "",
```
You can remove this, or fill it out to describe the cheat sheet

jsstrn replied:
I've removed it.

moollaza commented on share/goodie/cheat_sheets/json/html.json:
```diff
@@ -0,0 +1,340 @@
+    "Document metadata": [
+      {
+        "key": "<head>",
+        "val": "The HTML Head Element (<head>) provides general information (metadata) about the document, including its title and links to or definitions of scripts and style sheets"
```
I think this cheat sheet is a little too verbose. Te idea is to get the point across pretty quickly and somewhat bluntly. We can reduce the bulk of text be removing the content that repeats in all of these. i.e. "The HTML X Element (<X>) provides"

As well, let's just skip to the important info and be more succint."Provides metadata about the document including title, scripts and stylesheets".

The purpose of the "More at" link is to give the user more details and depth, so they can visit the link if they really want to know things in more detail.

jsstrn replied:
I see your point. I'll get to work on it. Thank you.

moollaza commented on share/goodie/cheat_sheets/json/html.json:
```diff
@@ -0,0 +1,340 @@
+        "val": "The HTML Table Head Element (<thead>) defines a set of rows defining the head of the columns of the table."
+      },
+      {
+        "key": "<tr>",
```
For the case where the element name isn't self explanatory we can do something like "Table Row. Defines a row of cells in a table."

jsstrn replied:
Good call.

jagtalon commented on share/goodie/cheat_sheets/json/html.json:
```diff
@@ -0,0 +1,340 @@
+      },
+      {
+        "key": "<script>",
+        "val": "     The HTML Script Element (<script>) is used to embed or reference an executable script within an HTML or XHTML document."
```
Remove the spaces at the beginning?

jagtalon commented:
Looks really useful!

jsstrn commented:
Improved this cheat sheet by rewording the element descriptions to make them less verbose. Also removed some elements that are not as widely used.

Thanks for the suggestions, guys! It's been a great first experience for me! Woot! More to come! 

Updated screenshots:
![screen shot 2015-08-30 at 4 38 48 pm](https://cloud.githubusercontent.com/assets/1199611/9566389/1b2135c2-4f36-11e5-819a-794823bf08a9.png)

![screen shot 2015-08-30 at 4 39 03 pm](https://cloud.githubusercontent.com/assets/1199611/9566391/2a679efe-4f36-11e5-8b85-d76969e8313b.png)

moollaza commented:
@jsstrn this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

BTW, for the future, we use 4-space indents :)

moollaza commented:
@jsstrn congrats! It's live: https://duckduckgo.com/?q=html%20cheat%20sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

moollaza commented:
By the way, we think you did a great job with this! Are you interested in helping others who build similar Instant Answers in the future?

If yes, we'll reach out and add you to a private list of mentors as well as send over some DDG merch as an additional thank you. Just email <open@duckduckgo.com> with the subject 'I want to help.' Please include your shipping address, phone-number (required for shipping outside USA) and t-shirt size. We'll reach out from there!

Otherwise, thank you for this amazing contribution. We hope you'll keep exploring and playing around with DuckDuckHack.
